### PR TITLE
How to Create ERC20 Tokens and Transact using Brownie (Python)

### LINKED ISSUE #6941: [Blockchain] Introduction on how to Create ERC20 Tokens and Transact using Brownie (Python)
### Topic Suggestion
Create ERC20 Tokens and Transact using Brownie (Python)

### Writing sample(s): 
https://www.section.io/engineering-education/sealed-classes-in-kotlin/
https://www.section.io/engineering-education/hilt-android-tutorial/

#### Proposed title of article
Create ERC20 Tokens and Transact using Brownie (Python)

#### Proposed article introduction
ERC20 tokens are a representation of something on the Ethereum Blockchain. They can reference a representation of Gold, in which case those tokens are very valuable. They can also reference a representation of someone's toenails, in which case those tokens are somewhat less valueable.

In this tutorial, you are going to do the following:
- Create your own ERC20 token
- Deploy your token to a local blockchain.
- Make transactions using your token
- Create a smart contract that interacts with your token.

#### Key takeaways 
- How to Create your own ERC20 token
- How to Deploy your token to a local blockchain.
- How to Make transactions using your token
- How to Create a smart contract that interacts with your token.

#### Article quality
Youtube videos that have been published on this are very shallow:
https://www.youtube.com/watch?v=8rpir_ZSK1g

They only teach users how to mint tokens. This article will dive deeper by teaching:
- How to check the balance
- How to transfer that token to another address
- What an allowance is and how to grant it. 

The docs are useful but they only describe the smart contracts. They do not describe how to handle this ERC20 tokens using  Brownie. They also don't explain how to Create a smart contract that interacts with your token.

#### References
https://docs.openzeppelin.com/contracts/2.x/api/token/erc20

### PR DESCRIPTION
**Pull request by RaphaelNdonga**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6941

### CODE REVIEW

**Comment by lalith1403** [comment]
plagiarism check done ✔️

File: content/articles/transacting-erc20-tokens-brownie/index.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+
+### Goal
+In this article, you will obtain all the knowledge you need to deploy and transact your own ERC20 token.
+
+### Table of contents
+1. Initialization.
+2. Deploy the Token.
+3. Transacting with the token.
+4. Creating a Smart Contract that transacts the Token
```
Comment (mohamedgh16) [inline content/articles/transacting-erc20-tokens-brownie/index.md:24]: Fix your table of contents so the reader can click and move directly to that section immediately, and add the missing headers. [Example](https://stackoverflow.com/questions/11948245/markdown-to-create-pages-and-table-of-contents).

File: content/articles/transacting-erc20-tokens-brownie/index.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+Your Token takes 2 parameters. `_founder` and `_initialSupply`. The founder will receive all the tokens from the initial supply.
+
+Your token inherits from OpenZeppelin's ERC20. It takes 2 parameters. `_name` and `_symbol`, which have been named `WILLIAM` AND `BILL`.
+
+Upon initialization, this contract grants the `_initialSupply` tokens to the `_founder`, through the `_mint` function.
+
+By default, this ERC20 token uses 18 decimals.
+
+Create a python file *transact_token.py* under the *scripts* folder
```
Comment (mohamedgh16) [inline content/articles/transacting-erc20-tokens-brownie/index.md:81]: A missing "."

File: content/articles/transacting-erc20-tokens-brownie/index.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+The account you will use to deploy this token will be `accounts[0]`. This is an address on a local blockchain network deployed using `ganache`. Brownie has `ganache` set up by default.
+
+In the terminal, run:
+
+```python
+brownie run transact_token
+```
+
+### Transacting with the Token
+You have your token deployed on the blockchain, now you need check the balance.
```
Comment (mohamedgh16) [inline content/articles/transacting-erc20-tokens-brownie/index.md:118]: Fix the spelling error here.

File: content/articles/transacting-erc20-tokens-brownie/index.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+There are a few functions which have not been covered here. For example:
+
+`_burn`. This is the opposite of `_mint`. It destroys the tokens in an account and reduces the total supply circulating.
+
+`decreaseAllowance`. This is the opposite of `increaseAllowance`. It reduces the tokens that an address is allowed to spend on an account.
+
+With the basics in place, you can easily adapt these functions into your code without a hassle.
+
+ERC20 tokens are very useful and have found applications in a variety of areas.
+Hopefully this article shed some light on how to deploy them and how to use them in transactions.
```
Comment (mohamedgh16) [inline content/articles/transacting-erc20-tokens-brownie/index.md:351]: Rephrase your conclusion, connect the sentences and explain to the reader what he gained from this tutorial.
Add a further reading section.

**Review by ghost — COMMENTED** [review]
Great article @RaphaelNdonga , implement the changes so we can move to the tech review.

**Comment by hectorkambow** [comment]
Fantastic 👍 

Please be sure to proofread the entire article as Peer Reviewers do not provide revisions services (and may not point out all pending issues).

If any other major edits are required @mohamedgh16 we can close the PR to allow the student the proper time to make the edits.

**Comment by mohamedgh16** [comment]
Great tutorial! @RaphaelNdonga , note the corrections to minimize revision next time.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**